File: smash-ranks-android/app/src/main/java/com/garpr/android/views/StringItemView.java
package com.garpr.android.views;

import android.content.Context;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;

import com.garpr.android.adapters.BaseAdapterView;

public class StringItemView extends AppCompatTextView implements BaseAdapterView<String> {

    public StringItemView(final Context context, final AttributeSet attrs) {
        super(context, attrs);
    }

    public StringItemView(final Context context, final AttributeSet attrs,
            final int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void setContent(final String content) {
        setText(content);
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/views/StringItemView.kt
package com.garpr.android.views

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.garpr.android.features.base.BaseAdapterView

class StringItemView @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet? = null
) : AppCompatTextView(context, attrs), BaseAdapterView<String> {

    override fun setContent(content: String) {
        text = content
    }

}
